.github/workflows/release.yaml

```diff
@@ -19,11 +19,55 @@ jobs:
         echo -e "# Insert Title\n" > Changes.md
-        git log --pretty=format:"%h -  %s - %an" ${{ github.event.inputs.tags }}..HEAD | grep -v "Merge pull" >> Changes.md
         echo -e "## Features\n\n## Fixes\n\n## Backwards incompatible changes\n\n## Docs\n\n## Misc\n\n## Thanks" >> Changes.md    - name: Draft Release
+        # this effectively gets the commit associated with github.event.inputs.tags
+        COMMON_ANCESTOR=$(git merge-base ${{ github.event.inputs.tags }} HEAD)
+        # in theory the github.event.inputs.release tag has not been created yet; do we want another input that specifies the existing
+        # commit desired for drafting the release? for now, we are using HEAD in the above git merge-base call
+
+        # use of 'hub', which is an extension of the 'git' CLI, allows for pulling of PRs, though we can't search based on commits
+        # associated with those PRs, so we grab a super big number, 300, which should guarantee grabbing all the PRs back to
+        # github.events.inputs.tags; we use grep -v to filter out release-note-none and release-note-action-required
+        hub pr list --state merged -L 300 -f "%sm;%au;%i;%t;%L%n" | egrep ", release-note|release-note," | grep -v release-note-none | grep -v release-note-action-required > last-300-prs-with-release-note.txt
+        # now we cylce through last-300-prs-with-release-note.txt, filtering out stuff that is too old or other anomalies,
+        # and update Changes.md with the release note.
+        while IFS= read -r pr; do
+          SHA=$(echo $pr | cut -d';' -f1)
+          # skip the common ancestor, which in essences is the commit associated with the tag github.event.inputs.tags
+          if [ "$SHA" == "$COMMON_ANCESTOR" ]; then
+             continue
+          fi
+
+          # styllistic clarification, purposefully avoiding slicker / cleverer / more compact scripting conventions
+
+          # this makes sure that this PR has merged
+          git merge-base --is-ancestor $SHA HEAD
+          rc=$?
+          if [ ${rc} -eq 1 ]; then
+             continue
+          fi
+          # otherwise, if the current commit from the last 300 PRs is not an ancestor of github.event.inputs.tags, we have gone too far, so skip
+          git merge-base --is-ancestor $COMMON_ANCESTOR $SHA
+          rc=$?
+          if [ ${rc} -eq 1 ]; then
+             continue
+          fi
+          # if we are at this point, we have a PR with a release note to add
+          AUTHOR=$(echo $pr | cut -d';' -f2)
+          PR_NUM=$(echo $pr | cut -d';' -f3)
+          PR_RELEASE_NOTE=$(wget -q -O- https://api.github.com/repos/shipwright-io/build/issues/${PR_NUM:1} | jq .body -r | grep -oPz '(?s)(?<=```release-note..)(.+?)(?=```)' | grep -avP '\W*(Your release note here|action required: your release note here|NONE)\W*')
+          echo -e "$PR_NUM by $AUTHOR: $PR_RELEASE_NOTE" >> Changes.md
+        done < last-300-prs-with-release-note.txt
```

Comment:
Can we move this to a bash script that we keep in-tree - example under `/hack`?

Reply:
Presumably ... just to be clear, what is motivating your ask here?

Reply:
Doing some research on how to do it, it seems perhaps that putting the shell script in the `.github` folder or `.github/actions` folder might be more "approved" or "best practice".

From some github dev forums:
- https://github.community/t/running-a-bash-script/141584
- https://github.com/actions/github-script

Reply:
+1 on externalizing this into a script.

Reply:
ok @adambkaplan @qu1queee I've moved the logic to a script file, following the github actions guidance from https://github.community/t/running-a-bash-script/141584